dashboard/src/modules/components/TableComponent/index.jsx

```diff
@@ -84,8 +92,13 @@ export const TableWithFavorite = () => {
   };
 
   const isRepoFavorited = (repo) => {
-    for (let i = 0; i < favoriteRepoNames.length; i++) {
-      if (repo.name === favoriteRepoNames[i].name) return true;
+    for (const element of favoriteRepoNames) {
+      if (
+        repo.name === element.name &&
+        repo.controller === element.controller &&
+        repo.metadata["dataset.created"] === element.metadata["dataset.created"]
+      )
```

Comment:
comparing only names will get the job done,since name of each and every dataset will be unique

Reply:
Changed `for` loop to `for-of` loop. Functionality remains the same

Reply:
Modified the code to check only for the controller name